src/Xrefcheck/Verify.hs

```diff
@@ -138,6 +139,35 @@ instance Buildable VerifyError where
         [h] -> ",\n   did you mean " +| h |+ "?\n"
         hs  -> ", did you mean:\n" +| blockListF' "    -" build hs
 
+data NeedsCaching key
+  = NoCaching
+  | CacheUnderKey key
+
+forConcurrentlyCaching
```

Comment:
Could you please add a haddock here describing what the function does and what we use it for?